### PR TITLE
[Sprint 3] Feature list order history logic

### PR DESCRIPTION
**Pull request by NguyenTienVinh**
Link redmine: https://dev.framgia.com/redmine/issues/114363
Task: List order history
Effect: N/A

### CODE REVIEW

File: app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryAdapter.java [hunk 1]
```diff
@@ -52,17 +58,17 @@ public Object getGroup(int groupPosition) {
...
     @Override
     public long getGroupId(int groupPosition) {
-        return 0;
+        return mOrderHistories.get(groupPosition).getId();
     }
 
     @Override
     public long getChildId(int groupPosition, int childPosition) {
-        return 0;
+        return mOrderHistories.get(groupPosition).getOrderDetails().get(childPosition).getId();
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryAdapter.java]: check ``OrderDetail != null``

File: app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryViewModel.java [hunk 2]
```diff
@@ -53,27 +80,40 @@ public OrderHistoryAdapter getOrderHistoryAdapter() {
         return mOrderHistoryAdapter;
     }
 
+    private void onCreateDatePickerDialog() {
+        new DatePickerDialog(mContext, this, mCalendar.get(Calendar.YEAR),
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryViewModel.java]: Please declare as a global variable & initialize in constructor

File: app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryAdapter.java [hunk 3]
```diff
@@ -42,7 +48,7 @@ public int getGroupCount() {
 
     @Override
     public int getChildrenCount(int groupPosition) {
-        return 0;
+        return mOrderHistories.get(groupPosition).getOrderDetails().size();
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryAdapter.java]: check ``OrderDetail != null``

File: app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryViewModel.java [hunk 4]
```diff
@@ -1,29 +1,56 @@
...
+    private OrderDetail mOrderDetail;
+    private Calendar mCalendar = Calendar.getInstance();
+
+    OrderHistoryViewModel(OrderHistoryAdapter orderHistoryAdapter, Navigator navigator,
+            Context context) {
         mOrderHistoryAdapter = orderHistoryAdapter;
         mNavigator = navigator;
+        mContext = context;
+    }
+    public OrderHistoryViewModel(Order order) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryViewModel.java]: remove this method

File: app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryViewModel.java [hunk 5]
```diff
@@ -1,29 +1,56 @@
...
+        mOrder = order;
     }
 
+    public OrderHistoryViewModel(OrderDetail orderDetail) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryViewModel.java]: remove this method

File: app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryViewModel.java (hunk 4 above)
Comment (NguyenTienVinh) [inline app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryViewModel.java]: em dùng bên adapter mà anh ,sao xóa đc ạ

File: app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryViewModel.java (hunk 5 above)
Comment (NguyenTienVinh) [inline app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryViewModel.java]: em dùng bên adapter mà anh ,sao xóa đc ạ

File: app/src/main/java/com/framgia/forder/screen/orderhistory/ItemHeaderOrderHistoryViewModel.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
+package com.framgia.forder.screen.orderhistory;
+
+import android.databinding.BaseObservable;
+import com.framgia.forder.data.model.Order;
+import com.framgia.forder.utils.Utils;
+
+import static com.framgia.forder.utils.Constant.StatusCode.ACCEPT_CODE;
+import static com.framgia.forder.utils.Constant.StatusCode.PENDING_CODE;
+import static com.framgia.forder.utils.Constant.StatusCode.REJECT_CODE;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/orderhistory/ItemHeaderOrderHistoryViewModel.java]: Please declare ``StatusCode`` in this file.

File: app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryAdapter.java [hunk 7]
```diff
@@ -52,16 +62,24 @@ public Object getGroup(int groupPosition) {
...
+        if (orderDetails != null) {
+            return mOrderHistories.get(groupPosition).getOrderDetails().get(childPosition);
+        }
         return null;
     }
 
     @Override
     public long getGroupId(int groupPosition) {
-        return 0;
+        return mOrderHistories.get(groupPosition).getId();
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryAdapter.java]: Please check ``mOrderHistories.get(groupPosition)!=null``

File: app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryAdapter.java [hunk 8]
```diff
@@ -52,16 +62,24 @@ public Object getGroup(int groupPosition) {
...
     }
 
     @Override
     public long getChildId(int groupPosition, int childPosition) {
+        List<OrderDetail> orderDetails = mOrderHistories.get(groupPosition).getOrderDetails();
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryAdapter.java]: Please check ``mOrderHistories.get(groupPosition)!=null``

File: app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryViewModel.java [hunk 9]
```diff
@@ -1,27 +1,47 @@
...
     private Navigator mNavigator;
     private String mStartDate;
     private String mEndDate;
-
-    OrderHistoryViewModel(OrderHistoryAdapter orderHistoryAdapter, Navigator navigator) {
+    private boolean mIsHidden;
+    private int mFlag;
+    private Context mContext;
+    private Calendar mCalendar;
+    private DatePickerDialog mDatePickerDialog;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryViewModel.java]: Please discuss with Tri to declare & using dialog in ``DialogManager`` file

File: app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryViewModel.java [hunk 10]
```diff
@@ -1,27 +1,47 @@
...
+
+    OrderHistoryViewModel(OrderHistoryAdapter orderHistoryAdapter, Navigator navigator,
+            Context context) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryViewModel.java]: context must the first param of method

File: app/src/main/java/com/framgia/forder/utils/Utils.java [hunk 11]
```diff
@@ -47,5 +49,15 @@ private static Date convertStringToDate(String date) {
                 return new Date();
             }
         }
+
+        public static String convertDateToString(Date date) {
+            SimpleDateFormat simpleDateFormat =
+                    new SimpleDateFormat(FORMAT_DATE, Locale.getDefault());
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/utils/Utils.java]: change ``FORMAT_DATE`` to ``Constant.FORMAT_DATE``

File: app/src/main/java/com/framgia/forder/screen/orderhistory/ItemHeaderOrderHistoryViewModel.java [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+import com.framgia.forder.utils.Utils;
+
+/**
+ * Created by ASUS on 28-04-2017.
+ */
+
+public class ItemHeaderOrderHistoryViewModel extends BaseObservable {
+    private static final int PENDING_CODE = 0;
+    private static final int ACCEPT_CODE = 1;
+    private static final int REJECT_CODE = 2;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/orderhistory/ItemHeaderOrderHistoryViewModel.java]: you can use java Anotation like ``Navigator.NavigateAnim`` and put it in utils package

File: app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryViewModel.java [hunk 13]
```diff
@@ -1,27 +1,62 @@
...
     private OrderHistoryContract.Presenter mPresenter;
     private OrderHistoryAdapter mOrderHistoryAdapter;
     private Navigator mNavigator;
     private String mStartDate;
     private String mEndDate;
-
-    OrderHistoryViewModel(OrderHistoryAdapter orderHistoryAdapter, Navigator navigator) {
+    private boolean mIsHidden;
+    private int mFlag;
+    private Context mContext;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryViewModel.java]: context must be declare in first variable, please move

File: app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryViewModel.java [hunk 14]
```diff
@@ -1,27 +1,62 @@
...
+    private DialogManager mDialogManager;
+
+    OrderHistoryViewModel(Context context, OrderHistoryAdapter orderHistoryAdapter,
+            Navigator navigator, DialogManager dialogManager) {
+        mContext = context;
         mOrderHistoryAdapter = orderHistoryAdapter;
         mNavigator = navigator;
+        mDialogManager = dialogManager;
+        mCalendar = Calendar.getInstance();
+        mDialogManager.dialogDatePicker(new DatePickerDialog.OnDateSetListener() {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/orderhistory/OrderHistoryViewModel.java]: Please ``implement DatePickerDialog.OnDateSetListener`` and change line to
```java
mDialogManager.dialogDatePicker(this);
```

File: app/src/main/java/com/framgia/forder/utils/OrderStatus.java [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+
+public class OrderStatus {
+
+    @IntDef({ PENDING_CODE, ACCEPT_CODE, REJECT_CODE })
+    public @interface StatusCode {
+        int PENDING_CODE = 0;
+        int ACCEPT_CODE = 1;
+        int REJECT_CODE = 2;
+    }
+}
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/utils/OrderStatus.java]: change this file to 
```java
@IntDef({ PENDING_CODE, ACCEPT_CODE, REJECT_CODE })
public @interface OrderStatusCode {
        int PENDING_CODE = 0;
        int ACCEPT_CODE = 1;
        int REJECT_CODE = 2;
}
```

File: app/src/main/java/com/framgia/forder/utils/OrderStatus.java [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+
+import static com.framgia.forder.utils.OrderStatus.OrderStatusCode.ACCEPT_CODE;
+import static com.framgia.forder.utils.OrderStatus.OrderStatusCode.PENDING_CODE;
+import static com.framgia.forder.utils.OrderStatus.OrderStatusCode.REJECT_CODE;
+
+/**
+ * Created by ASUS on 28-04-2017.
+ */
+
+public class OrderStatus {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/utils/OrderStatus.java]: rename file to ``OrderStatusCode``

**Pull request merged**